paddle/fluid/imperative/tracer.cc

```diff
@@ -188,10 +190,16 @@ void Tracer::TraceOp(const std::string& type, const NameVarMap<VarType>& ins,
   NameVarMap<VarType> new_ins = ins;
   if (amp_level_ == AmpLevel::O1) {
     VLOG(5) << "Auto mixed precision run operator: " << type;
-    new_ins = AutoCastInputs<VarType>(type, ins);
+    if (amp_dtype_ == AmpDtype::D1) {
+      new_ins = AutoCastInputs<VarType>(type, ins);
+    }
   } else if (amp_level_ == AmpLevel::O2) {
     VLOG(5) << "Pure fp16 run operator: " << type;
-    new_ins = CastPureFp16Inputs<VarType>(type, ins);
+    if (amp_dtype_ == AmpDtype::D1) {
+      new_ins = CastPureFp16Inputs<VarType>(type, ins);
+    } else if (amp_dtype_ == AmpDtype::D2) {
+      new_ins = CastPureBf16Inputs<VarType>(type, ins);
+    }
```

Comment:
why bf16 only supports o2?

Reply:
supported O1：op in white list will use bf16, other will use fp32.